packages/celotool/src/lib/geth.ts

```diff
@@ -511,7 +564,9 @@ export const simulateClient = async (
 
   while (true) {
     const sendTransactionTime = Date.now()
+    await unlock(kit, kit.defaultAccount, password, 9223372036, unlockNeeded)
```

Comment:
I would put the `if` from `unlock` here rather than inside `unlock`, move `unlockNeeded = false` into the if, and remove `unlockNeeded` as a parameter to `unlock`.